### ISSUE TITLE
Error while executing command: findElement: Unknown locator strategy id at 

### ISSUE THREAD

**Issue by davewood** [teodesian/Selenium-Remote-Driver#319]
Selenium::Remote::Driver 1.20
FF 53.0.3
Linux Debian Jessie

```
use Selenium::Firefox;
my $driver = Selenium::Firefox->new(
   marionette_enabled => 1,
   binary => "/home/david/dev/EPPlication/geckodriver",
   firefox_binary => "/home/david/dev/EPPlication/firefox/firefox",
   desired_capabilities => { overlappingCheckDisabled => 1 },
);
$driver->get("https://chrysalis.lab.foo.net");
$element = $driver->find_element("_c_0_","id")->click;
$driver->find_element(q{//input[@name="name"]})->send_keys("demouser-6370912591");
# submit form
$driver->find_element("_save_","id")->click();
$driver->pause(1000);
```

Use of uninitialized value in hash element at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/ErrorHandler.pm line 126.
Error while executing command: findElement: Unknown locator strategy id at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/Driver.pm line 327.
 at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/Driver.pm line 327.

**Comment by davewood** [comment]
$resp->status is not defined, here is a dump of $resp

$driver->find_element_by_id("_c_0_")

```
2017-06-09 10:05:12 5168 $VAR1 = {
          'value' => { 
                       'error' => 'invalid argument',
                       'message' => 'Unknown locator strategy id',
                       'stacktrace' => 'stack backtrace:
   0:           0x4fa94d - backtrace::backtrace::trace::h45ace4059cd74233
   1:           0x4fae32 - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
   2:           0x4f55ea - webdriver::error::WebDriverError::new::hcee37049f67bbb3d
   3:           0x4f5541 - webdriver::common::LocatorStrategy::from_json::h507c3065aa985d7f
   4:           0x4e95d0 - <webdriver::command::LocatorParameters as webdriver::command::Parameters>::from_json::hb14d97d2c1215db1
   5:           0x41a397 - <webdriver::command::WebDriverMessage<U>>::from_http::h599719aac1d99fcb
   6:           0x429b07 - <webdriver::server::HttpHandler<U> as hyper::server::Handler>::handle::h464126b9de8bccc4
   7:           0x406d83 - std::panicking::try::do_call::h26a4a61aaab84e76
   8:           0x5b77fa - panic_unwind::__rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-musl-linux/build/src/libpanic_unwind/lib.rs:98
   9:           0x4170ff - <F as alloc::boxed::FnBox<A>>::call_box::h5bc17fbe43f7212f
  10:           0x5b0014 - alloc::boxed::{{impl}}::call_once<(),()>
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-musl-linux/build/src/liballoc/boxed.rs:624
                         - std::sys_common::thread::start_thread
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-musl-linux/build/src/libstd/sys_common/thread.rs:21   
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-musl-linux/build/src/libstd/sys/unix/thread.rs:84'
                     }
        };
Use of uninitialized value in hash element at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/ErrorHandler.pm line 129.
Error while executing command: findElement: Unknown locator strategy id at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/Driver.pm line 327.
 at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/Driver.pm line 327.
 at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/Finders.pm line 23.
Can't locate object method "click" via package "0" (perhaps you forgot to load "0"?) at -e line 16.
```

**Comment by davewood** [comment]
something is broken here 
I managed to find elements by xpath but sending keys gave another cryptic error msg

> $driver->find_element( q{//input[@name="name"]} )->send_keys("demouser-123");

Error while executing command: sendKeysToElement: Missing 'text' parameter at /home/david/perl5/perlbrew/perls/perl-5.22.2/lib/site_perl/5.22.2/Selenium/Remote/Driver.pm line 327.

**Comment by gerhardj** [comment]
We're also having some issues with geckodriver. Our solutions so far:

* instead of `$driver->find_element("myname", "name")` we use `$driver->find_element("[name=myname]",css)`
* instead of `$driver->find_element("myid","id")` we use `$driver->find_element("#myid","css")`
* and for send_keys we have a little patch in place as mika referenced above.

**Comment by gempesaw** [comment]
yeah, `name` is not accepted by geckodriver as a valid find method. technically nothing to do with S::R::D, but on the other hand this issue keeps coming up, and it's not too hard to switch a `name` find over to a css one. but, i'm not sure it's appropriate...

**Comment by gempesaw** [comment]
hummm, i think i need to look into these errors a bit more. 321 inadvertently closed this automatically

**Comment by teodesian** [comment]
This is likely a dupe of #345.

**Comment by teodesian** [comment]
This was indeed a dupe.  Fixed by #345.
